src/rust/ide/view/graph-editor/src/component/breadcrumbs.rs

```diff
@@ -181,15 +223,33 @@ impl BreadcrumbsModel {
         let screen     = camera.screen();
         let x_position = -screen.width/2.0;
         let y_position = screen.height/2.0;
-        self.root.set_position(Vector3(x_position.round(),y_position.round(),0.0));
+        self.set_position(Vector3(x_position.round(), y_position.round(), 0.0));
     }
 
-    fn width(&self) -> f32 {
+    fn breadcrumbs_container_size(&self) -> f32 {
         self.breadcrumbs.borrow().iter().map(|breadcrumb| breadcrumb.width()).sum()
     }
 
-    fn relayout_for_project_name_width(&self, width:f32) {
-        self.breadcrumbs_container.set_position_x(width.round());
+    fn update_layout(&self) {
+        let is_macos     = ARGS.platform.map(|p|p.is_macos()) == Some(true);
+        let is_frameless = ARGS.frame == Some(false);
+        let x_offset     = if is_macos && is_frameless { MACOS_TRAFFIC_LIGHTS_WIDTH }
```

Comment:
Why did you move this logic into breadcrumbs? I don't think this component is responsible for making the spacing on the left side. We will use breadcrumbs in the future in multiple panels, not only on top of the window. So to correctly position breadcrumbs is the responsibility of its parent, not this component itself

Reply:
Because in my implementation, the breadcrumb object stretched to contain the traffic lights. Also, I found the positioning a bit weird as it was before. The breadcrumbs seemed to position themselves in the corner of the screen, but we added this offset from the outside. It just isn't what I would expect.